src/workflow-engine/charms/temporal-worker/oci_factory/notification/mattermost_notifier.py

```diff
@@ -0,0 +1,93 @@
+import json
+import logging
+import os
+import requests
+import sys
+from typing import Optional
+
+
+logging.basicConfig(level=os.environ.get("TWC_LOG_LEVEL", "info").upper())
+
+COLORS = {
+    "success": "#33CC33",
+    "failure": "#CC0000",
+    "unknown": "#B0B0B0",
+}
+
+REQUIRED_ENVS = ["MATTERMOST_CHANNEL_ID", "MATTERMOST_TOKEN", "MATTERMOST_SERVER"]
+
+missing_envs = [env for env in REQUIRED_ENVS if os.environ.get(env) is None]
+if missing_envs:
+    logging.error(
+        f"Unable to notify on Mattermost, missing environment variables: {missing_envs}"
+    )
+    sys.exit(1)
+
+MATTERMOST_CHANNEL_ID = os.environ.get("MATTERMOST_CHANNEL_ID")
+MATTERMOST_TOKEN = os.environ.get("MATTERMOST_TOKEN")
+MATTERMOST_SERVER = os.environ.get("MATTERMOST_SERVER")
+
+POST_URL = f"{MATTERMOST_SERVER}/api/v4/posts"
+PATCH_URL = POST_URL + "/{}/patch"
+
+HEADERS = {
+    "Authorization": f"Bearer {MATTERMOST_TOKEN}",
+    "Content-Type": "application/json",
+}
+
+
+def send_message(title: str, message: str) -> str:
+    payload = {
+        "channel_id": MATTERMOST_CHANNEL_ID,
+        "props": {
+            "attachments": [
+                {
+                    "fallback": message,
+                    "title": title,
+                    "text": message,
+                    "color": COLORS["unknown"],
+                }
+            ]
+        },
+    }
+
+    token = os.environ.get("MATTERMOST_TOKEN", None)
```

Comment:
why this? you have all in MATTERMOST_TOKEN

Reply:
Good catch. Removed.